ios/CodePush/CodePushDownloadHandler.m

```diff
@@ -98,13 +112,24 @@ - (void)connection:(NSURLConnection*)connection didFailWithError:(NSError*)error
 }
 
 - (void)connectionDidFinishLoading:(NSURLConnection *)connection {
+    [self.outputFileStream close];
+    
+    // MUST have received bytes from remote server. do not do this, we will get a 0 bytes file of download.zip in download directory,
+    // the variable isZip is FALSE and download.zip will be renamed app.bundle that occurring by self.doneCallback(isZip) method.
+    if (self.receivedContentLength < 1) {
+        NSError *err = [NSError errorWithDomain:@"received nothing"
+                                           code:-1
+                                       userInfo:nil];
+        self.failCallback(err);
```

Comment:
If `didReceiveResponse` gets called above, will `connectionDidFinishLoading` get called again after? Since we already called the `failCallback` once before, we don't want to call it again, because we will end up rejecting the promise to JS twice which will result in an error.